### PR TITLE
Show feedback for setting Z-Wave JS config parameters

### PR DESCRIPTION
**Pull request by cgarwood**
## Proposed change
Show feedback on the Z-Wave JS device config page when parameters are updated:

image

image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts [hunk 1]
```diff
@@ -369,6 +416,18 @@ class ZWaveJSNodeConfig extends SubscribeMixin(LitElement) {
     return [
       haStyle,
       css`
+        .accepted {
+          color: green;
+        }
+
+        .queued {
+          color: #fca503;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts:424]: Use the variables we have for this: 
```
      --error-color: #db4437;
      --warning-color: #FF9800;
      --success-color: #0f9d58;
      --info-color: #4285f4;
```

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts [hunk 2]
```diff
@@ -321,20 +354,34 @@ class ZWaveJSNodeConfig extends SubscribeMixin(LitElement) {
...
+        this.hass,
+        this.configEntryId!,
+        nodeId!,
+        target.property,
+        value,
+        target.propertyKey ? target.propertyKey : undefined
+      );
+      this._config![target.key].value = value;
+
+      this.setResult(target.key, result.status);
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts:374]: We set the value in the config, and then set the result to config? I don't get this? I think we should combine these in the set result function?

File: src/data/zwave_js.ts [hunk 3]
```diff
@@ -29,10 +29,15 @@ export interface ZWaveJSNode {
...
 export interface ZWaveJSNodeConfigParams {
+  [key: string]: ZWaveJSNodeConfigParam;
+}
+export interface ZWaveJSNodeConfigParam {
   property: number;
   value: any;
   configuration_value_type: string;
   metadata: ZWaveJSNodeConfigParamMetadata;
+  result?: string;
+  error?: string;
```
Comment (bramkragten) [inline src/data/zwave_js.ts:40]: We should not store this in `ZWaveJSNodeConfigParam`, as that is what we get from the backend. If we would refresh it, we lose our data. Create a new object.

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts (hunk 1 above)
Comment (cgarwood) [reply src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts:424]: Aha, I wondered if something like that existed. Is there a list of variables somewhere?

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts [hunk 4]
```diff
@@ -178,6 +188,12 @@ class ZWaveJSNodeConfig extends SubscribeMixin(LitElement) {
   }
 
   private _generateConfigBox(id, item): TemplateResult {
+    const icons = {
+      accepted: mdiCheckCircle,
+      queued: mdiProgressClock,
+      error: mdiCloseCircle,
+    };
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts:195]: Move this out of the class, it never changes

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts [hunk 5]
```diff
@@ -212,7 +212,7 @@ class ZWaveJSNodeConfig extends SubscribeMixin(LitElement) {
               )}
             </em>`
           : ""}
-        ${result
+        ${result && result.status
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts:215]:
```ts
        ${result?.status
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**